### PR TITLE
Add options to drain command, and force reboot

### PR DESCRIPTION
**Pull request by timothyrr**
We have encountered scenarios in our environment where the reboot never happens due to hanging pods or a misconfigured PDB that does not allow a reboot. This adds a few new options to kured:

`--drain-grace-period`: This gets passed to the drain command as: `--grace-period` with the same default as in the kubernetes docs.
`--drain-timeout`: This is the total amount of time to wait before forcing a reboot.
`--force-drain`: This is the same old `--force` option that gets passed to the drain command, but now you can choose whether to enable or disable that option (previously it was always enabled).
`--force-reboot`: Tells kured whether to force it to reboot after a specified timeout no matter the status of the drain command.

### CODE REVIEW

**Comment by dholbach** [comment]
@bboreham As you had a comment on #79 - how do you feel about this solution?

**Comment by dholbach** [comment]
Just a quick PSA from me:

- we're hanging out in #kured on Slack
- we're planning a contributors meeting soon: here's the [link to the doodle](https://doodle.com/poll/tuwg4w4mxbbknhe3) to find a good time - be sure to pick your timezone before submitting ([preliminary agenda](https://docs.google.com/document/d/1bsHTjHhqaaZ7yJnXF6W8c89UB_yn-OoSZEmDnIP34n8/edit#))

We'd love to see you there.

**Comment by andrsp** [comment]
It would be nice to have it merged and released :)

**Comment by sebikul** [comment]
Would it be possible to also include a fix to issue https://github.com/weaveworks/kured/issues/121 in this PR. We are hitting a race condition with our Load Balancer draining the requests from the node, but it rebooting too fast for this process to finish.

I'm not a Go programmer, but if someone could review my code I'm more than happy to try to implement this, and whatever is needed to merge this PR.

**Comment by timothyrr** [comment]
I am going to work on the fixes mentioned by @bboreham this week. I can look at that #121 issue as well. I think that should be easy enough to add in.

File: cmd/kured/main.go [hunk 1]
```diff
@@ -236,11 +249,36 @@ func drain(nodeID string) {
...
-	if err := drainCmd.Run(); err != nil {
+	if err := drainCmd.Start(); err != nil {
 		log.Fatalf("Error invoking drain command: %v", err)
 	}
+
+	done := make(chan error)
+	go func() { done <- drainCmd.Wait() }()
+
+	if forceReboot == true {
+	    select {
```
[the remark this answers is not in the document]
Comment (timothyrr) [reply cmd/kured/main.go:262]: @bboreham can you clarify what you mean by this exactly? I am not sure if I completely follow what should be moving with the timeouts? I have been following timeout examples I have seen here: https://gobyexample.com/timeouts

[the remark this answers is not in the document]
Comment (timothyrr) [reply cmd/kured/main.go]: @bboreham can you take another look? I believe it is now a bit more simplified without having 2 selects. Not sure if this is along the lines of what you were thinking or not.

**Comment by dholbach** [comment]
We recently put in https://github.com/weaveworks/kured/commit/1ba3acab98fa217ea65405640706c55c89be956d so it should be easier to implement now?

**Review by dholbach — CHANGES_REQUESTED** [review]
Needs work after https://github.com/weaveworks/kured/commit/1ba3acab98fa217ea65405640706c55c89be956d and #210

**Comment by timothyrr** [comment]
I am working on this now and will hopefully have changes made by early next week.

**Comment by dholbach** [comment]
Sorry. Can you please retarget your branch to be merged into `main` please?

**Pull request closed without merging**